lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -126,18 +126,27 @@ static bool readOptionsBlock(llvm::BitstreamCursor &cursor,
     case options_block::XCC:
       extendedInfo.addExtraClangImporterOption(blobData);
       break;
-    case options_block::PLUGIN_SEARCH_PATH:
-      extendedInfo.addPluginSearchPath(blobData);
-      break;
-    case options_block::EXTERNAL_SEARCH_PLUGIN_PATH:
-      extendedInfo.addExternalPluginSearchPath(blobData);
-      break;
-    case options_block::COMPILER_PLUGIN_LIBRARY_PATH:
-      extendedInfo.addCompilerPluginLibraryPath(blobData);
-      break;
-    case options_block::COMPILER_PLUGIN_EXECUTABLE_PATH:
-      extendedInfo.addCompilerPluginExecutablePath(blobData);
+    case options_block::PLUGIN_SEARCH_OPTION: {
+      unsigned optKind;
+      StringRef optStr;
+      options_block::ResilienceStrategyLayout::readRecord(scratch, optKind);
+      switch (PluginSearchOptionKind(optKind)) {
+      case PluginSearchOptionKind::PluginPath:
+        optStr = "-plugin-path";
```

Comment:
Not sure about this. But I didn't wanted to introduce another "kind" enum just for `swift/Serialization/Validation.h`

Reply:
The fact that LLDB needs to match on these strings feels a little brittle. Would we be better off exporting `PluginSearchOptionKind` into a header and having that in the list of plugin search options?

Reply:
👍 Added a commit.
Made `PluginSearchOption` a struct with `PluginSearchOption::Kind` in `SearchPathOptions.h`